### DISCUSSION TITLE
Conflicting file name: `layouts/index.json`

### DISCUSSION THREAD

**Discussion by FuadEfendi** [razonyang/hugo-theme-bootstrap#726] [Q&A] [answered]
Hello,

I tried to create separate JSON file to upload to Algolia.
I followed the guide at https://forestry.io/blog/search-with-algolia-in-hugo/ (which has minor issues such as missing [OUTPUTS] section in config):
```
[outputs]
home = ["HTML", "RSS", "JSON", "Algolia"]

[outputFormats.Algolia]
baseName = "algolia"
isPlainText = true
mediaType = "application/json"
notAlternative = true

[params.algolia]
vars = ["title", "summary", "date", "publishdate", "expirydate", "permalink"]
params = ["categories", "tags"]
```

File `layouts/_default/list.algolia.json`:
```
{{/* Generates a valid Algolia search index */}}
{{- $.Scratch.Add "index" slice -}}
{{- $section := $.Site.GetPage "section" .Section }}
{{- range .Site.AllPages -}}
  {{- if or (and (.IsDescendant $section) (and (not .Draft) (not .Params.private))) $section.IsHome -}}
    {{- $.Scratch.Add "index" (dict "objectID" .UniqueID
"fuzzywordcount" .FuzzyWordCount
"keywords" .Keywords
"permalink" .Permalink
"title" .Title "type" .Type
"url" .URL
"section" .Section "tags" .Params.Tags "categories" .Params.Categories )}}
  {{- end -}}
{{- end -}}
{{- $.Scratch.Get "index" | jsonify -}}
```

Whatever I do, newly created `public/algolia.json` is **exactly the same as `public/index/index.json`**

Finally, I realized that using name `layout/index.json` is wrong, because probably it defines "template" for other JSON files (such as my `algolia.json`); we probably breach some naming conventions.

The issue disappeared only when I started using whole local copy of theme and made changes directly to the theme. I started seeing what "algolia" template supposed to produce.

I think `index.json` name is special for Hugo; but not sure, only few weeks of experience. 

Thanks,

**Comment by razonyang** [comment, marked as answer]
Oh, sorry about that, I forgot the [**Hugo look up order**](https://gohugo.io/templates/lookup-order/) totally.

In this case, you'll need to place your template to `layouts` folder to get higher priority.

Move the JSON template to `layouts/home.algolia.json` should works.

And the template syntax seems outdated, can't build, I modified a bit to make site buildable.

```html
{{/* Generates a valid Algolia search index */}}
{{- $.Scratch.Add "index" slice -}}
{{- $section := $.Site.GetPage "section" .Section }}
{{- range .Site.AllPages -}}
  {{- if or (and (.IsDescendant $section) (and (not .Draft) (not .Params.private))) $section.IsHome -}}
    {{- $.Scratch.Add "index" (dict "objectID" .File.UniqueID
"fuzzywordcount" .FuzzyWordCount
"keywords" .Keywords
"permalink" .Permalink
"title" .Title "type" .Type
"url" .Permalink
"section" .Section "tags" .Params.Tags "categories" .Params.Categories )}}
  {{- end -}}
{{- end -}}
{{- $.Scratch.Get "index" | jsonify -}}
```

![image](https://user-images.githubusercontent.com/17720932/191130850-dc8a4d5b-a965-4fbb-9ae4-3710ef64eee0.png)

Hope this helps.
